app/models/king.rb

```diff
@@ -12,6 +12,77 @@ def valid_move?(x, y)
     return false
   end
 
+  def queenside_valid?(x, y)
+    if x_space - 2 == x and
+      move_count == nil and
+      y_space == y 
+      if color = "black"
+        queenside_rook = Piece.find(x_space: 0, y_space: 0, type: "Rook", color: "black", move_count: nil)
+        if queenside_rook != nil and
+          !is_obstructed?(0, 0)
+          return true
+        end
+      end
+      if color = "white"
+        queenside_rook = Piece.find(x_space: 0, y_space: 7, type: "Rook", color: "white", move_count: nil)
+        if queenside_rook != nil and
+          !is_obstructed?(0, 7)
+          return true
+        end
+      end
+    end
+    
+    return false
+  end
+
+  def kingside_valid?(x, y)
+    if x_space + 2 == x and
+      move_count == nil and
+      y_space == y 
+      if color = "black"
+        kingside_rook = Piece.find(x_space: 7, y_space: 0, type: "Rook", color: "black", move_count: nil)
+        if kingside_rook != nil and
+          !is_obstructed?(7, 0)
```

Comment:
@BrandonJohnson65, thanks again for reminding me about obstructions! Let me know if there's anything else that looks funky.

Reply:
Looks good to me!